### ISSUE TITLE
On iOS if you dismiss the gui keyboard once with enter the engine reports a constant flow of key_enter input events

### ISSUE THREAD

**Issue by subsoap** [defold/editor2-issues#2745]
![2019-09-18 02_30_37-Faerie Solitaire Harvest - Defold Editor 2 0](https://user-images.githubusercontent.com/409170/65136382-54206900-d9bc-11e9-998c-6a4d5aaab902.png)
![2019-09-18 02_29_22-Faerie Solitaire Harvest - Defold Editor 2 0](https://user-images.githubusercontent.com/409170/65136384-54206900-d9bc-11e9-84ee-be9b8e8e7a45.png)

**Comment by subsoap** [comment]
Happens with auto hide true and false

gui.show_keyboard(gui.KEYBOARD_TYPE_DEFAULT, false) 

To be clear it's the "return" virtual key that gets infinite input

![2019-09-18 02_43_25-ipad virtual keyboard - Google Search](https://user-images.githubusercontent.com/409170/65137392-1cb2bc00-d9be-11e9-820a-c201f43cdf74.png)

**Comment by subsoap** [comment]
It is caused if you manually do gui.hide_keyboard() when detecting enter event, if you let keyboard close itself it doesn't happen.

[iOSKeyReturn.zip](https://github.com/defold/editor2-issues/files/3625741/iOSKeyReturn.zip)

**Comment by subsoap** [comment]
Only happens on input if you delay gui.hide_keyboard with a message for example it's not a problem.

**Comment by subsoap** [comment]
Ok, I was wrong, the enter key STILL gets stuck if you press return at all, it doesn't matter if you delay using gui.hide_keyboard.

**Comment by subsoap** [comment]
Worth noting it is happening on iPad Retina. Not on iPhone 5.

**Comment by subsoap** [comment]
After more testing it seems like a delay via messages working or not working is related to autohide being true or not.

**Comment by subsoap** [comment]
Auto hide false = delay works. Auto hide true = delay doesn't help to prevent.

**Comment by britzl** [comment]
Thanks. I've opened an issue in our engine backlog to track this.

**Comment by britzl** [comment]
I can reproduce the problem with the enter key (and backspace) never generating a release if the keyboard is programmatically hidden as a result of a press event for enter or backspace.

But when fixing this it uncovers some issues in your example.

```
function on_input(self, action_id, action)
	if action_id == hash("key_enter") then
		self.key_enter = self.key_enter + 1
		toggle_keyboard(self)
	end
	if action.released and gui.pick_node(gui.get_node("box"), action.x, action.y) then
		toggle_keyboard(self)
	end
end
```

There should be a check for action.released (or pressed) on the enter key and there should be a check for action_id "touch" on the gui.pick_node() call:

```
function on_input(self, action_id, action)
	if action_id == hash("key_enter") and action.pressed then
		self.key_enter = self.key_enter + 1
		toggle_keyboard(self)
	end
	if action_id == hash("touch") and action.released and gui.pick_node(gui.get_node("box"), action.x, action.y) then
		toggle_keyboard(self)
	end
end

```

**Comment by subsoap** [comment]
I see!

https://github.com/Lerg/extension-ios_keyboard_notification/archive/master.zip

This is what I've been using in our projects.

**Comment by britzl** [comment]
This was released in 1.2.164
